### ISSUE TITLE
on_policy_distillation, advantage_estimator

### ISSUE THREAD

**Issue by EthanChen1234** [THUDM/slime#1449]
Hi, thank you for your great work.

I have a question regarding the advantage estimator in on-policy-distillation. In the current implementation, is appears as:
`advantages = [
            teacher_log_prob - student_log_prob
            for teacher_log_prob, student_log_prob in zip(teacher_log_probs, student_log_probs, strict=False)
        ]`

However, in the MiMo-V2-Flash Technical Report（https://arxiv.org/abs/2601.02780），I noticed that the advantages seem to be detached, which suggests a formulation like:
`advantages = [
            (teacher_log_prob - student_log_prob)**.detach()**
            for teacher_log_prob, student_log_prob in zip(teacher_log_probs, student_log_probs, strict=False)
        ]`
I'm a bit confused about it, could you please clarify the correct approach?

**Comment by ahxt** [comment]
based on the thinking machines lab implementation, the student_log_prob is not detached. based on my experiments, whether it is detached or not does not matter, and this is also supported by a prior paper.
